SauveJeanLuc/coffee-tasks
# Make coffee task a progressive web application

## Description
- Added vite-plugin-PWA to devdependencies of coffee-task application. Vite-plugin-PWA automatically creates a service worker.
- Added PWA manifest requirements which include logo assets in the required sizes and file type
- Added the install button on the homepage to render if the user has not installed the coffee-task application


## Related Issue
Closes #75 





## Motivation and Context
I am currently learning the benefits and implementation of PWAs and thought what better way to learn than getting feedback from other developers? The Coffee tasks application is one that would have a better user experience if the user can just click on the application from either their home screen or desktop screen depending on the device used to visit the website. 



## How Has This Been Tested?
To test the application in development :
- Add the code below to the variable pwaOptions in the vite.config.js file
```
devOptions: {
enabled: true,
navigateFallBack: 'index.html',
},
mode: 'development',
``` 
- Run ` yarn dev` in your terminal. The server can now be accessed from localhost in your browser.
- Open the developer tool in your browser and generate the website report using lighthouse.
![Screenshot from 2022-10-09 22-06-03](https://user-images.githubusercontent.com/24270269/194779485-86a027e6-4d2f-4fc1-a563-8562b1e4eb9e.png)
- The PWA check is all passed



# Checklist:


## Screenshots (if appropriate):
